test/integration/targets/canonical-pep517-self-packaging/runme.sh

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env bash
+
+if [[ "${ANSIBLE_DEBUG}" == true ]]  # `ansible-test` invoked with `--debug`
+then
+    CLI_VERBOSE_FLAG=-v
+    SET_DEBUG_MODE=-x
+else
+    ANSIBLE_DEBUG=false
+    CLI_VERBOSE_FLAG=
+    SET_DEBUG_MODE=+x
+fi
+
+
+set -eEuo pipefail
+
+source virtualenv.sh
+
+set "${SET_DEBUG_MODE}"
+
+
+CLEANED_UP=false
+
+DIST_NAME=ansible_core
+
+LOWEST_SUPPORTED_BUILD_DEPS_FILE="${PWD}"/minimum-build-constraints.txt
+
+SRC_ROOT_DIR="$(dirname "$(dirname "$(dirname "$(dirname "${OUTPUT_DIR}")")")")"
+GENERATED_MANPAGES_SUBDIR=docs/man/man1/
+
+function git-clean-manpages() {
```

Comment:
Instead of modifying the source tree, why not use a copy instead? That way there's no risk of missing changes, and no need for any cleanup logic at all. The `test/results/` directory would just need to be excluded from the copy.

Reply:
Honestly, I'd be more precise to do this with a Git checkout.

Also, what if one invoked build locally and now, there's `docs/man/man1/` on disk in their project checkout? We still need to clean that too.

Reply:
Is there a relatively simple way we can include the generated man pages in the sdist without first writing them to the source tree?

Reply:
@mattclay yes and @nitzmahone even posted a snippet @ https://github.com/ansible/ansible/pull/79606#discussion_r1081941187. I'd like that too.

I was originally trying to convert what the cascade of the makefile targets do as close as possible, and consciously avoided trying to refactor it all along the way to prevent introducing invisible behavior changes.
How would you feel about doing this in a follow-up PR? I'm trying to keep my PRs as atomic as possible.